addons/web/controllers/main.py

```diff
@@ -1046,7 +1046,7 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas
             filename_field=filename_field, download=download, mimetype=mimetype,
             default_mimetype='image/png', access_token=access_token)
 
-        if status == 301 or (status != 200 and download):
+        if status in [301, 304] or (status != 200 and download):
```

Comment:
Maybe use a set instead of a list here, apparently it is much faster to check if an element is `in`. I never tested the actual impact though.

Reply:
Ok I tested it, and while a big set is much faster to look into, as I suspected, creating it here would take more time in the first place. Also frozenset is always better than set. And list seems to be just a bit better than tuple.

So list it is, no change required.